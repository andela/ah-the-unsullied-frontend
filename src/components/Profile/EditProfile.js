import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Row, Modal, Input } from 'react-materialize';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import * as profileActions from '../../actions/profile/profile';
import ImageUploaoder from '../../utils/ImageUploaoder';
import '../../assets/styles/HomePage.scss';

class ViewProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bio: "",
      image: props.image
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit = e => {
    e.preventDefault();
    const { bio, image } = this.state;
    const data = {
      bio,
      image
    };
    const { actions } = this.props;
    actions.EditUserProfile('sammy', data);
  };

  fileHandler = e => {
    const selectFile = e.target.files[0];
    ImageUploaoder({ image: selectFile }).then(res =>
      this.setState({
        image: res.data.secure_url
      })
    );
  };

  onChange(e) {
    this.setState({
      bio: e.target.value
    });
  }

  render() {
    return (
      <div>
        <Modal
          header="Profile Edit"
          className="editPage"
          fixedFooter
          trigger={
            <button className="btn waves-effect edit">
              Edit profile
              <i className="material-icons right">edit</i>
            </button>
          }
        >
          <Row>
            <img
              src={this.props.image}
              className="responsive-img circle editPageProfile"
              onClick={() => this.fileInput.click()}
              height="100px"
              width="100px"
              alt="Avatar"
            />
          </Row>
          <Row>
            <form onSubmit={this.handleSubmit}>
              <label htmlFor="bio">Bio</label>
              <textarea
                id="bio"
                className="materialize-textarea"
                onChange={this.onChange}
                defaultValue={this.props.bio}
              />
              <Row>
                <Input
                  s={12}
                  type="file"
                  name="image"
                  ref={fileInput => {
                    this.fileInput = fileInput;
                  }}
                  label="Upload Image"
                  onChange={this.fileHandler}
                  defaults={12}
                />
              </Row>
              <Row className="left-align">
                <button className="btn waves-effect edit " type="submit">
                  Update
                </button>
              </Row>
            </form>
          </Row>
        </Modal>
      </div>
    );
  }
}

ViewProfile.propTypes = {
  actions: PropTypes.func.isRequired,
  bio: PropTypes.string.isRequired
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(profileActions, dispatch)
  };
}

function mapStateToProps(state) {
  return {
    UserProfile: state.profile
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewProfile);
