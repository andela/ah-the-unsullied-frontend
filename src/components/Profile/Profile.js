import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Col, Preloader, Row } from 'react-materialize';
import ProfileView from './views/viewProfileView';
import * as profileActions from '../../actions/profile/profileActions';

class Profile extends Component {
  state = {
    fetched: false
  };

  componentDidMount = () => {
    const { actions } = this.props;
    const username = this.props.auth.user.username;

    actions.getUserProfile(username).then(() => {
      this.setState({
        fetched: true
      });
    });
  };

  render() {
    if (!this.state.fetched) {
      return (
        <div>
          <Row>
            <Col s={12}>
              <Preloader size="big" />
            </Col>
          </Row>
        </div>
      );
    } else {
      const { profile } = this.props.UserProfile;

      return (
        <div>
          <ProfileView profile={profile} />
        </div>
      );
    }
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(profileActions, dispatch)
  };
}

function mapStateToProps(state) {
  return {
    UserProfile: state.profile,
    auth: state.auth
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profile);
