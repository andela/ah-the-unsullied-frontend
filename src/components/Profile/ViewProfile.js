import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Row, Modal, Input } from 'react-materialize';
import * as profileActions from '../../actions/profile/profile';
import ImageUploaoder from '../../utils/ImageUploaoder';
import '../../assets/styles/HomePage.scss';

class ViewProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bio: '',
      image: ''
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit = e => {
    e.preventDefault();
    const { bio, image } = this.state;
    console.log(this.state);
    const data = {
      bio,
      image
    };
    const { actions } = this.props;
    actions.EditUserProfile('sammy', data);
  };

  fileHandler = e => {
    const selectFile = e.target.files[0];
    ImageUploaoder({ image: selectFile }).then(res =>
      this.setState({
        image: res.data.secure_url
      })
    );
  };

  onChange(e) {
    const { name } = e.target;
    this.setState({
      bio: e.target.value
    });
  }

  render() {
    return (
      <div>
        <Modal
          className="editPage"
          fixedFooter
          trigger={
            <button className="btn waves-effect edit">Edit profile</button>
          }
        >
          <Row>
            <img
              src={this.props.image}
              className="responsive-img circle editPageProfile"
              onClick={() => this.fileInput.click()}
              height="150px"
              width="150px"
            />
          </Row>
          <Row>
            <form onSubmit={this.handleSubmit}>
              <textarea
                s={12}
                label="Bio"
                name="bio"
                onChange={this.onChange}
                validate
                defaultValue={this.props.bio}
              />
              <Input
                type="file"
                ref={fileInput => {
                  this.fileInput = fileInput;
                }}
                onChange={this.fileHandler}
                defaults={12}
              />
              <Input type="submit" />
            </form>
          </Row>
        </Modal>
      </div>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(profileActions, dispatch)
  };
}

function mapStateToProps(state) {
  return {
    UserProfile: state.profile
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewProfile);
