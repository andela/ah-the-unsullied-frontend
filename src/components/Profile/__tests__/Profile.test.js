import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import { Profile, mapStateToProps } from '../Profile';

Enzyme.configure({ adapter: new Adapter() });
describe('get profile', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },

      profile: {
        profile: {}
      },
      match: {
        params: {
          username: 'nesh'
        }
      },

      getUserProfile: async () => {
        await jest.fn();
      }
    };

    state = {
      auth: {},
      profile: {},
      fetched: true
    };

    wrapper = shallow(<Profile {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should mapStateToProps', () => {
    const props = mapStateToProps(state);
    expect(props.profile).toEqual(state.profile.profile);
  });

  it('should render article view component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
