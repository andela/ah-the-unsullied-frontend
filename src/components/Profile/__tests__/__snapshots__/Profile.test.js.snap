// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get profile should render article view component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    getUserProfile={[Function]}
    match={
      Object {
        "params": Object {
          "username": "nesh",
        },
      }
    }
    profile={
      Object {
        "profile": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row>
        <Col
          s={12}
        >
          <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />
        </Col>
      </Row>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          s={12}
        >
          <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />
        </Col>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Preloader
            active={true}
            color="blue"
            flashing={false}
            size="big"
          />,
          "s": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "color": "blue",
            "flashing": false,
            "size": "big",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row>
          <Col
            s={12}
          >
            <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />
          </Col>
        </Row>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            s={12}
          >
            <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />
          </Col>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Preloader
              active={true}
              color="blue"
              flashing={false}
              size="big"
            />,
            "s": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "color": "blue",
              "flashing": false,
              "size": "big",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
